.github/workflows/test.yml

```diff
@@ -0,0 +1,61 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - "ubuntu-latest"
+          - "macos-latest"
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: make build
+      - name: Test
+        run: |
+          go install gotest.tools/gotestsum@latest
+          gotestsum --format testname --junitfile ${{ matrix.platform }}-junit-report.xml
+      - uses: actions/upload-artifact@v3
+        if: success() || failure()
+        with:
+          name: test-results
+          path: '*-junit-report.xml'
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - name: lint-prep
+        run: go mod tidy && git diff --exit-code
+      - name: lint-run
+        run: make lint
+      - name: Go vet
```

Comment:
**todo**: opened an https://github.com/elastic/apm-aws-lambda/issues/367 for this. We should probably make sure go vet is run as part of the lint task but we can iterate on it on a separate PR.